src/msg/async/AsyncConnection.cc

```diff
@@ -2083,7 +2083,8 @@ int AsyncConnection::send_message(Message *m)
   } else {
     out_q[m->get_priority()].emplace_back(std::move(bl), m);
     ldout(async_msgr->cct, 15) << __func__ << " inline write is denied, reschedule m=" << m << dendl;
-    center->dispatch_event_external(write_handler);
+    if (can_write != WriteStatus::REPLACING)
+      center->dispatch_event_external(write_handler);
```

Comment:
okay, iiuc, once we are at `STATE_CONNECTING_READY`, the pending outgoing message will be dispatched.